clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
@hachikuji I just had a discussion with @becketqin @onurkaraman @radai-rosenblatt and we think it is more intuitive to keep the current behavior. Maybe we should discuss this in open source mailing list before you recover the previous behavior. Do you want me to open a thread? What do you think?